core-web/libs/edit-content/src/lib/utils/functions.util.ts

```diff
@@ -61,5 +65,9 @@ export const getFinalCastedValue = (value: string | undefined, field: DotCMSCont
         return value?.split(',').map((value) => value.trim());
     }
 
+    if (field.fieldType === FIELD_TYPES.BLOCK_EDITOR) {
+        return JSON.parse(JSON.stringify(value));
+    }
```

Comment:
We need a test for this, there's two ways of test this, in theory this should be a different reference from the original object you can use that to test it or find a way to mock both `JSON` methods